test/common/network/addr_family_aware_socket_option_impl_test.cc

```diff
@@ -24,8 +23,9 @@ TEST_F(AddrFamilyAwareSocketOptionImplTest, SetOptionFailure) {
 // If a platform supports IPv4 socket option variant for an IPv4 address, it works
 TEST_F(AddrFamilyAwareSocketOptionImplTest, SetOptionSuccess) {
   Address::Ipv4Instance address("1.2.3.4", 5678);
-  const int fd = address.socket(Address::SocketType::Stream);
-  EXPECT_CALL(socket_, fd()).WillRepeatedly(Return(fd));
+  IoHandlePtr io_handle = address.socket(Address::SocketType::Stream);
+  ScopedIoHandleCloser closer(io_handle);
```

Comment:
I doubt that we need this closer here and below as fd wasn't explicitly closed before this change. But I could have missed something. Do you mind add ASSERT(fd_ != -1) in IoSocketHandle::close() to verify that.

Reply:
That assert is already there- @mattklein123 already suggested it. As I recall (I'll check again), the assert in IoSocketHandle::close() fired without this.

Reply:
I just verified that we need this ScopedIoHandleCloser- the test asserts without it.

Reply:
This sounds more like something to do in the next phase of the PR, when the socket-related syscalls are moved into the IoHandle. When that is implemented, it would be a bug to have isolated close(fd) calls that are not implemented through IoHandle::close(). Do you agree?

Reply:
I'm totally fine to leave it to next PR.